src/solution.c

```diff
@@ -429,6 +434,56 @@ static void sol_thd_sleep(systime_t *deadline, systime_t interval)
   chSysUnlock();
 }
 
+/** Calculates Iono and Tropo correction
+ * \param n_ready_tdcp The number of measurements written to `m_tdcp`
+ * \param nav_meas_tdcp Pointer to measurements array
+ * \param pos_ecef position in WGS84 Earth Centered, Earth Fixed (ECEF) Cartesian
+ * \param pos_llh position in WGS84 geodetic coordinates*/
+void calc_iono_tropo(u8 n_ready_tdcp,
+                     navigation_measurement_t *nav_meas_tdcp,
+                     double *pos_ecef,
+                     double *pos_llh)
+{
+  for (u8 i = 0; i < n_ready_tdcp; i++) {
+    ionosphere_t iono_params;
+    double az, el, cp_correction;
+    double tropo_correction = 0;
+    double iono_correction = 0;
+    /* calculate azimuth and elevation of SV */
+    wgsecef2azel(nav_meas_tdcp[i].sat_pos, pos_ecef, &az, &el);
+    /* read iono parameters for the tracking channel if available */
+    if (gps_iono_params_read(&iono_params)) {
+      /* calculate iono correction */
+      iono_correction = calc_ionosphere(&nav_meas_tdcp[i].tot,
+                                        pos_llh[0],
+                                        pos_llh[1],
+                                        az,el,
+                                        &iono_params);
+      /* correct pseudorange */
```

Comment:
Please, note:
This change depends on PR https://github.com/swift-nav/libswiftnav-private/pull/6 where following constants and functions defined: GPS_L2_HZ, code_to_carr_freq()